diff --git a/src/Components/Components/src/ComponentFactory.cs b/src/Components/Components/src/ComponentFactory.cs
index 0618c9a4d2..8f09324ee7 100644
--- a/src/Components/Components/src/ComponentFactory.cs
+++ b/src/Components/Components/src/ComponentFactory.cs
@@ -62,7 +62,7 @@ internal sealed class ComponentFactory
 
     public IComponent InstantiateComponent(IServiceProvider serviceProvider, [DynamicallyAccessedMembers(Component)] Type componentType, IComponentRenderMode? callerSpecifiedRenderMode, int? parentComponentId)
     {
-        var (componentTypeRenderMode, propertyInjector) = GetComponentTypeInfo(componentType);
+        var (componentTypeRenderMode, _) = GetComponentTypeInfo(componentType);
         IComponent component;
 
         if (componentTypeRenderMode is null && callerSpecifiedRenderMode is null)
@@ -89,16 +89,7 @@ internal sealed class ComponentFactory
 
         if (!_propertyInjectionDisabled)
         {
-            if (component.GetType() == componentType)
-            {
-                // Fast, common case: use the cached data we already looked up
-                propertyInjector(serviceProvider, component);
-            }
-            else
-            {
-                // Uncommon case where the activator/resolver returned a different type. Needs an extra cache lookup.
-                PerformPropertyInjection(serviceProvider, component);
-            }
+            PerformPropertyInjection(serviceProvider, component);
         }
 
         return component;
